### PR TITLE
@damassi => Auction2 show all mediums and enable add to calendar on hover

### PR DESCRIPTION
**Pull request by sweir27**
Fixing up a few of the remaining issues for the auction2 page! (https://github.com/artsy/auctions/issues/310)

This PR updates the `AddToCalendar` component to show up on _hover_ instead of click.

hover-cal

It also updates the `MediumFilter` to show _all_ mediums (based on the same `medium_map` that we use on the `/collect` page). Mediums that do not show up in the sale artwork aggregation are disabled/grayed out, but do not disappear.

Note that I decided to reuse the `/collect` medium map, since both the `ArtworkFilterQuery` and `SaleArtworkFilterQuery` aggregate based on the same mediums (https://github.com/artsy/gravity/blob/master/app/models/search/queries/artwork_filtered_query.rb#L10). Basically we only return values in our aggregation that are included in that array of genes, and for the UI we exclude `Installation` and `Performance Art` from the filter list.

image

### CODE REVIEW

File: desktop/apps/auction2/components/medium_filter/index.js [hunk 1]
```diff
@@ -23,9 +26,16 @@ function MediumFilter(props) {
         checked={allMediumsSelected}
       />
       {
-        aggregatedMediums.map((agg) => {
-          const mediumSelected = contains(mediumIds, agg.id)
-          return <BasicCheckbox key={agg.id} item={agg} onClick={updateMediumParamsAction} checked={mediumSelected} />
+        _.map(mediumMap, (name, id) => {
+          const mediumSelected = contains(mediumIds, id)
+          const includedMedium = aggregatedMediums.find((agg) => { return agg.id === id })
```
Comment (damassi) [inline desktop/apps/auction2/components/medium_filter/index.js]: I would shorten to `.find(agg => agg.id === id)` unless you have a pref for this format?

File: desktop/apps/auction2/components/medium_filter/index.js [hunk 2]
```diff
@@ -23,9 +26,16 @@ function MediumFilter(props) {
...
+          return <BasicCheckbox
+                    key={id}
+                    item={{id, name, count: includedMedium && includedMedium.count}}
+                    onClick={updateMediumParamsAction}
+                    checked={mediumSelected}
+                    disabled={includedMedium === undefined}
+                  />
```
Comment (damassi) [inline desktop/apps/auction2/components/medium_filter/index.js]: minor, but

```javascript
return (
  <BasicCheckbox
    key={id}
    item={{id, name, count: includedMedium && includedMedium.count}}
    ... 
  />
)
```
to keep things consistent.

File: package.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
...
+    "test": "yarn mocha test/lib && yarn mocha $(find desktop/test -name '*.coffee') && yarn mocha $(find desktop/components/*/test -name '*.coffee') && yarn mocha $(find desktop/components/**/*/test -name '*.coffee') && yarn mocha $(find desktop/apps/*/test -name '*.coffee') && yarn mocha $(find desktop/apps/*/test -name '*.js') && yarn mocha $(find desktop/apps/*/**/*/test -name '*.coffee') && yarn mocha $(find mobile/test -name '*.coffee') && yarn mocha $(find mobile/components/*/test -name '*.coffee') && yarn mocha $(find mobile/components/**/*/test -name '*.coffee') && yarn mocha $(find mobile/apps/*/test -name '*.coffee') && yarn mocha $(find mobile/apps/*/**/*/test -name '*.coffee')",
```
Comment (damassi) [inline package.json]: Not totally sure if you can search by regex but we could do `*.{js,coffee}`

Comment (sweir27) [inline package.json]: Oh! That's great. Yeah, added this as a bandaid since I realized that none of the new tests were running in CI :sob:

**Comment by damassi** [comment]
Looks good, will merge on 🍏

**Pull request merged**